integration-tests/narayana-lra/src/test/java/org/acme/quickstart/lra/LRAParticipantTest.java

```diff
@@ -0,0 +1,124 @@
+
+        assertEquals(HttpURLConnection.HTTP_CREATED, response.getStatusCode());
+        // the id of the new LRA is in the Location header:
+        String lraId = response.getHeaders().get("Location").getValue();
+        assertNotNull(lraId, "invocation did not return an LRA");
+
+        // read the status of the LRA. Since the LRA should have ended the coordinator will no longer know about it
+        String lraStatus = coordinatorGetRequest(String.format("%s/status", lraId),
+                new int[] { HttpURLConnection.HTTP_NOT_FOUND, HttpURLConnection.HTTP_OK }, false);
+
+        if (lraStatus != null) {
+            assertEquals(LRAStatus.Closed.name(), lraStatus, "LRA should have closed");
+        }
+    }
+
+    // test that JAX-RS ServerLRAFilter can start and end an LRA in separate business methods
+    @Test
+    public void testLRA() throws IOException {
+        Response response = given().when().post("/start");
+
+        assertEquals(HttpURLConnection.HTTP_CREATED, response.getStatusCode());
+        // the id of the new LRA is in the Location header:
+        String lraId = response.getHeaders().get("Location").getValue();
+        assertNotNull(lraId, "invocation did not return an LRA");
+
+        // verify that the coordinator knows about the new LRA (the last component is the transaction uid
+        String[] segments = lraId.split("/");
+        assertNotEquals(0, segments.length);
+        String uid = segments[segments.length - 1];
+
+        String allLRAs = coordinatorGetRequest(coordinatorEndpoint, new int[] { HttpURLConnection.HTTP_OK }, true);
+        assertNotNull(allLRAs);
+        assertTrue(allLRAs.contains(uid));
+
+        // read the status of the LRA. Since the LRA hasn't ended yet the coordinator should still know about it
+        String lraStatus = coordinatorGetRequest(String.format("%s/%s/status", coordinatorEndpoint, uid),
+                new int[] { HttpURLConnection.HTTP_OK }, true);
+        assertEquals(LRAStatus.Active.name(), lraStatus, "LRA should still be active");
+
+        // now call a method that should end the LRA:
+        response = given().header(LRA_HTTP_CONTEXT_HEADER, lraId).when().put("/end");
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+        assertEquals(lraId, response.getBody().print());
+
+        // read the status of the LRA. Since the LRA should have ended the coordinator will no longer know about it
+        lraStatus = coordinatorGetRequest(String.format("%s/status", lraId),
+                new int[] { HttpURLConnection.HTTP_NOT_FOUND, HttpURLConnection.HTTP_OK }, false);
+
+        if (lraStatus != null) {
+            assertEquals(LRAStatus.Closed.name(), lraStatus, "LRA should have closed");
+        }
+    }
+
+    // send an HTTP GET request to an endpoint
+    private static String coordinatorGetRequest(String endpoint, int[] expectedResponseCodes, boolean readResponse)
```

Comment:
Why not using restassured for this? You can pass absolute URLs too.

Reply:
The routine uses Jupiter:
```java
assertTrue(Arrays.stream(expectedResponseCodes).anyMatch(i -> i == responseCode),
                "unexpected response code: GET " + endpoint);
```